src/hellogo_fares/read.rs

```diff
@@ -0,0 +1,1120 @@
+                    .map(|id| id == stop_point_ref)
+                    .unwrap_or(false)
+            })
+            .collect();
+        if selected_scheduled_stop_points.len() != 1 {
+            bail!(
+                "Failed to find a unique 'ScheduledStopPoint' with reference '{}'",
+                stop_point_ref
+            )
+        }
+        let scheduled_stop_point = selected_scheduled_stop_points[0];
+        fn remove_netex_prefix<'a>(reference: &'a str) -> Result<&'a str> {
+            if let Some(index) = reference.find(":") {
+                if reference.len() > index + 1 {
+                    Ok(&reference[index + 1..])
+                } else {
+                    bail!("Failed to remove prefix from '{}'", reference)
+                }
+            } else {
+                bail!("Failed to find ':' to remove a prefix in '{}'", reference)
+            }
+        }
+        let stop_point_ids = scheduled_stop_point
+            .try_only_child("projections")?
+            .children()
+            .filter(|element| element.name() == "PointProjection")
+            .flat_map(|point_projection| point_projection.children())
+            .filter(|element| element.name() == "ProjectedPointRef")
+            .flat_map(|projected_point_ref| projected_point_ref.attr("ref"))
+            .map(|reference| remove_netex_prefix(reference))
+            .collect::<Result<_>>()?;
+        Ok(stop_point_ids)
+    }
+    let start_stop_point_ids = get_stop_point_ids(scheduled_stop_points, &start_stop_point_ref)?;
+    let end_stop_point_ids = get_stop_point_ids(scheduled_stop_points, &end_stop_point_ref)?;
+    fn get_stop_point_from_collections<'a>(
+        collections: &'a Collections,
+        stop_point_id: &str,
+        prefix: &str,
+    ) -> Option<&'a StopPoint> {
+        collections
+            .stop_points
+            .get(&format!("{}:{}", prefix, stop_point_id))
+    }
+    let start_stop_area_ids: BTreeSet<_> = start_stop_point_ids
+        .iter()
+        .flat_map(|stop_point_id| {
+            get_stop_point_from_collections(collections, stop_point_id, prefix)
+        })
+        .map(|stop_point| stop_point.stop_area_id.clone())
+        .collect();
+    let end_stop_area_ids: BTreeSet<_> = end_stop_point_ids
+        .iter()
+        .flat_map(|stop_point_id| {
+            get_stop_point_from_collections(collections, stop_point_id, prefix)
+        })
+        .map(|stop_point| stop_point.stop_area_id.clone())
+        .collect();
+    let origin_destinations = start_stop_area_ids
+        .iter()
+        .flat_map(|origin| {
+            end_stop_area_ids
+                .iter()
+                .map(move |destination| (origin.clone(), destination.clone()))
+        })
+        .collect();
+    Ok(origin_destinations)
+}
+
+fn load_fare_file(collections: &mut Collections, root: &Element) -> Result<()> {
+    let prefix = get_prefix(&collections)?;
+    let prefix_column = prefix.clone() + ":";
```

Comment:
Only concatenate if a prefix exists, otherwise the result of the variable does not make sense (and objects in the collection will not be found).

Reply:
Do you mean something like this.
```suggestion
    let prefix_column = if prefix.is_empty() {
    	"".to_string()
    } else {
    	prefix.clone() + ":"
    };
```

Reply:
Oh, I actually made the `get_prefix()` function used just above return a `Result`, meaning, it would fail if no prefix is found. Maybe I need to convert to an `Option`. Let me try to rework all of that.

Reply:
So I discussed the edge cases with @prhod and:
- If `contributor_id` is `NTM:` (so without a suffix), just act as normal and extract the prefix
- If `contributor_id` is something like `123` (no prefix), process the NTFS as a non-prefixed NTFS

I also realized that the name `prefix_column` didn't make much sense because... what I meant was `prefix_with_colon`. I also did the change accordingly. Look commit 599ab461779fa74b7d1dde216e1e8621d4427836 for the changes.